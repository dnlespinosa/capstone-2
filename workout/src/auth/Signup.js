import React, { useContext, useState } from 'react';
import { useNavigate } from 'react-router-dom';

const Signup = ({ signup }) => {
    const history  = useNavigate()
    const [formData, setFormData] = useState({
        username: '', 
        password: '',
        firstName: '', 
        lastName: '',
        email: ''
    })

    async function handleSubmit(e) {
        e.preventDefault();
        let res = await signup(formData)
        if (res.success) {
            history('/your-program')
        } else {
            console.log(res.errors)
        }
    }

    function handleChange(evt) {
        const { name, value } = evt.target;
        setFormData(data => ({ ...data, [name]: value}))
    }

    return (
        <>
            <div style={{color: 'white'}}>
            <h2>Signup Form</h2>

            <form onSubmit={handleSubmit}>
                <label>Username</label>
                <input 
                    name='username'
                    value={formData.username}
                    onChange={handleChange} 
                />

                <label>Password</label>
                <input 
                    name='password'
                    value={formData.password}
                    onChange={handleChange} 
                />

                <label>First Name</label>
                <input 
                    name='firstName'
                    value={formData.firstName}
                    onChange={handleChange} 
                />

                <label>Last Name</label>
                <input 
                    name='lastName'
                    value={formData.lastName}
                    onChange={handleChange} 
                />

                <label>Email</label>
                <input 
                    name='email'
                    value={formData.email}
                    onChange={handleChange} 
                />

                <button type='submit' onSubmit={handleSubmit}>Sign Up!</button>
            </form>
            </div>
        </>
    )
}

export default Signup