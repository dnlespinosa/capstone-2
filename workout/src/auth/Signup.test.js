import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import Signup from "./Signup";
import { UserProvider } from "../testUtils";


it("matches snapshot", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <UserProvider>
          <Signup />
        </UserProvider>
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});