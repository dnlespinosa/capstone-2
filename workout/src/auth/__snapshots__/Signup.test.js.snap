// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <h2>
    Signup Form
  </h2>
  <form>
    <label>
      Username
    </label>
    <input
      name="username"
      value=""
    />
    <label>
      Password
    </label>
    <input
      name="password"
      value=""
    />
    <label>
      First Name
    </label>
    <input
      name="firstName"
      value=""
    />
    <label>
      Last Name
    </label>
    <input
      name="lastName"
      value=""
    />
    <label>
      Email
    </label>
    <input
      name="email"
      value=""
    />
    <button
      type="submit"
    >
      Sign Up!
    </button>
  </form>
</DocumentFragment>
`;
